app/javascript/packs/components/restaurants/restaurants.jsx

```diff
@@ -0,0 +1,78 @@
+import React from 'react';
+import { connect } from 'react-redux';
+
+import { searchRestaurants } from '../../reducers/restaurants/actions';
+import {
+    getRestaurants,
+    isLoading,
+    isLoaded,
+} from '../../reducers/restaurants/selectors';
+
+import RestaurantsLayout from './views/restaurantsLayout';
+import MapPlaceholder from './views/mapPlaceholder';
+import ListNullstate from './views/listNullstate';
+import ListLoading from './views/listLoading';
+import RestaurantsHeader from './restaurantsHeader';
+import RestaurantsList from './restaurantsList';
+
+class Restaurants extends React.Component {
+    renderList() {
+        if (!this.props.isLoaded) {
+            return (
+                <ListLoading />
+            );
+        } else if (this.props.restaurants.length > 0) {
+            return (
+                <RestaurantsList restaurants={this.props.restaurants} />
+            );
+        } else {
+            return (
+                <ListNullstate />
+            );
+        }
+    }
+
+    renderMap() {
+        return (
+            <MapPlaceholder/>
+        );
+    }
+
+    renderHeader() {
+        return (
+            <RestaurantsHeader search={this.props.search} />
+        );
+    }
+
+    render() {
+        return (
+            <RestaurantsLayout
+                header={this.renderHeader()}
+                map={this.renderMap()}
+                list={this.renderList()}
+            />
+        );
+    }
+
+    componentDidMount() {
+        this.props.search();
+    }
+}
+
+const mapStateToProps = (state) => {
+    return {
+        restaurants: getRestaurants(state),
+        isLoading: isLoading(state),
+        isLoaded: isLoaded(state),
+    };
+};
+
+const mapDispatchToProps = (dispatch) => {
+    return {
+        search: (filters = {}, isInBackground = false) => {
+            dispatch(searchRestaurants(filters, isInBackground));
+        },
+    };
+};
```

Comment:
You can write this in Redux shorthand:
```js
const mapDispatchToProps = {
  search: searchRestaurants,
};
```

Reply:
I can't I set a default values in function params

Reply:
They're all passed as `...args`

Reply:
```javascript
search: (filters = {}, isInBackground = false) => {
            dispatch(searchRestaurants(filters, isInBackground));
        },
```
If I change it, than `search()` won't work, instead I will have to pass `search({}, false)`

Reply:
Oh. In that case, yeah, you can't use shorthand.
Unless you add the default parameters to `searchRestaurants` itself.